autoload/sj/elm.vim

```diff
@@ -0,0 +1,247 @@
+" vim: foldmethod=marker
+
+" Split / Join {{{1
+"
+" function! sj#elm#Join() {{{2
+"
+" Attempts to join the multiline tuple, record or list closest
+" to current curspor position.
+function! sj#elm#Join()
+  let [from, to] = sj#elm#ClosestMultilineBraces()
+
+  if from[0] < 0
+    return 0
+  endif
+
+  let original = sj#GetByPosition([0, from[0], from[1]], [0, to[0], to[1]])
+  let joined = sj#elm#JoinOuterBraces(sj#elm#JoinLines(original))
+  call sj#ReplaceByPosition([0, from[0], from[1]], [0, to[0], to[1]], joined)
+
+  return 1
+endfunction
+
+" function! sj#elm#Split() {{{2
+"
+" Attempts to split the tuple, record or list fursest to current
+" curspor position on the same line.
+function! sj#elm#Split()
+  call sj#PushCursor()
+
+  let [from, to] = sj#elm#CurrentLineOutermostBraces(col('.'))
+
+  if from < 0
+    return 0
+  endif
+
+  let parts = sj#elm#SplitParts(from, to)
+
+  if len(parts) <= 2
+    return 0
+  endif
+
+  let replacement = join(parts, "\n")
+
+  call cursor(line('.'), from)
+  let [previousLine, previousCol] = searchpos('\S', 'bn')
+  if previousLine == line('.') && previousCol > 0 && sj#elm#CharAt(previousCol) =~ '[=:]'
+    let replacement = "\n".replacement
+    let from = previousCol + 1
+  end
+
+  call sj#ReplaceCols(from, to, replacement)
+
+  call sj#PopCursor()
+
+  return 1
+endfunction
+
+
+" Helper functions {{{1
+
+" function! sj#elm#SkipSyntax() {{{2
+"
+" Returns Elm's typical skippable syntax (strings and comments)
+function! sj#elm#SkipSyntax()
+  return sj#SkipSyntax(['elmString', 'elmTripleString', 'elmComment'])
+endfunction
+
+" function! sj#elm#CurrentLineClosestBraces(column) {{{2
+"
+" Returns the columns corresponding to the bracing pair closest
+" to and surrounding given column as a list ([opening, closing]).
+" Returns [-1, -1] when there is none on the same line.
+function! sj#elm#CurrentLineClosestBraces(column)
+  call sj#PushCursor()
+
+  let skip = sj#elm#SkipSyntax()
+  let currentLine = line('.')
+
+  call cursor(currentLine, a:column)
+  let from = searchpairpos('[{(\[]', '', '[})\]]', 'bcn', skip, currentLine)
+
+  if from[0] == 0
+    call sj#PopCursor()
+
+    return [-1, -1]
+  end
+
+  call cursor(from[0], from[1])
+  normal! %
+
+  if line('.') != currentLine
+    call sj#PopCursor()
+
+    return [-1, -1]
+  endif
+
+  let to = col('.')
+
+  call sj#PopCursor()
+
+  return [from[1], to]
+endfunction
```

Comment:
You might consider a `try-finally` for the cursor push/pop here as well. You've correctly popped the cursor in all the returns, but it might simplify the code.